### PR TITLE
Provide fallbacks for KeyFile name and comment when absent or blank

### LINKED ISSUE #141: Missing descriptions of custom commands in Startup tab.
## What Happened
The added custom startup command shows empty.
Screenshot from 2020-11-10 00-58-23

## Expected Behavior
The command text should be shown.

## Platform Information
elementary OS 5.1.7 Hera

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/97359816-missing-descriptions-of-custom-commands-in-startup-tab?utm_campaign=plugin&utm_content=tracker%2F56794370&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F56794370&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by jeremypw**
Fixes #141 although custom commands should ideally provide their own name and comment -unclear whether this can be enforced - requires a separate PR.

### CODE REVIEW

**Comment by danrabbit** [comment]
@jeremypw can you resolve conflicts with master?

**Comment by jeremypw** [comment]
Conflicts resolved

File: src/Startup/Backend/KeyFile.vala [hunk 1]
```diff
@@ -166,9 +166,22 @@ public class Startup.Backend.KeyFile : GLib.Object {
     }
 
     public Entity.AppInfo create_app_info () {
+        var key_name = keyfile_get_string (KeyFileDesktop.KEY_NAME).strip ();
+        if (key_name == null || key_name == "") {
+            key_name = keyfile_get_string (KeyFileDesktop.KEY_EXEC).strip ();
+            if (key_name == null || key_name == "") {
+                key_name = _("Command");
```
Comment (danrabbit) [inline src/Startup/Backend/KeyFile.vala:173]: It looks like the default name that gets set for this is:
```suggestion
                key_name = _("Custom Command");
```

**Review by danrabbit — COMMENTED** [review]
So you've found something kind of weird I think!

It looks like for custom commands we special case them somehow. We're not using the `name` and `comment` keys in the list in the same way. But we probably should!

So if that were the case, then the `name` key should be set the same as `exec` and comment should be set as `path` as you have it here

File: src/Startup/Backend/KeyFile.vala (hunk 1 above)
Comment (jeremypw) [reply src/Startup/Backend/KeyFile.vala]: Done

**Comment by jeremypw** [comment]
This is only a stop-gap really. There are a number of deficiencies in the handling of custom startup commands should anyone be moved to address them.

**Comment by marbetschar** [comment]
I'm no longer able to reproduce on Odin - so I guess this has been fixed and can be closed (along with the corresponding issue)?

Screenshot from 2021-09-01 08-23-04

**Comment by jeremypw** [comment]
@marbetschar OK, thanks for checking it out!

**Pull request closed without merging**